webflows/src/main/java/com/schibsted/account/webflows/client/Client.kt

```
@@ -274,18 +275,22 @@ class Client : ClientInterface {
 
 data class OAuthError(val error: String, val errorDescription: String?) {
     companion object {
-        fun fromJson(json: String): OAuthError {
-            val parsed = JSONObject(json)
-            return OAuthError(
-                parsed.getString("error"),
-                parsed.optString("error_description")
-            )
+        fun fromJson(json: String): OAuthError? {
+            try {
+                val parsed = JSONObject(json)
+                return OAuthError(
+                    parsed.getString("error"),
+                    parsed.optString("error_description")
+                )
+            } catch (e: JSONException) {
+                return null
```

If my understanding is correct, in case of HTML error response from backend (which is common) this null will be returned, and we'll have 0 information about error cause. Could we at least return an error code in such case? OAuthError could be an sealed class - either error/errorDescription or just code?

---

Thanks for the input!

The result of this is not passed further anywhere (and there's no info here about actual HTTP status code, etc.). The actual response is already propagated in the two places where it is used:
1. [Client.handleAuthenticationResponse](https://github.com/schibsted/account-sdk-android-web/blob/3ce2455e6b832e0602cd6f9be0bf151c1be970e8/webflows/src/main/java/com/schibsted/account/webflows/client/Client.kt#L190)
2. [User.refreshTokens](https://github.com/schibsted/account-sdk-android-web/blob/3ce2455e6b832e0602cd6f9be0bf151c1be970e8/webflows/src/main/java/com/schibsted/account/webflows/user/User.kt#L189)

So it should already be treated as an error, just that now it will work as expected since the JSON parsing won't throw an exception.